src/Stack/Options.hs

```diff
@@ -279,6 +280,12 @@ configOptsParser hide0 =
            <> help "Extra directories to check for libraries"
            <> hide
             )))
+    <*> optionalFirst (textOption
+             ( long "with-ghc"
```

Comment:
I just cargo culted that - I'm happy to make the change, although I think it should be done categorically, as it's "not the way it's currently done".

Thoughts?